key_gen.go

```diff
@@ -57,78 +57,62 @@ func KeyForPrefixedStringMap(
 	return keyForPrefixedStringMaps(prefix, stringMap)
 }
 
-// keyForPrefixedStringMaps generates a unique key for a prefix and a series
-// of maps containing tags.
-//
-// If a key occurs in multiple maps, keys on the right take precedence.
-func keyForPrefixedStringMaps(prefix string, maps ...map[string]string) string {
-	keys := keyGenPool.stringsPool.Get().([]string)
+func keyForPrefixedStringMapsAsKey(buf []byte, prefix string, maps ...map[string]string) []byte {
+	keys := stringsPool.Get().(*[]string)
 	for _, m := range maps {
 		for k := range m {
-			keys = append(keys, k)
+			*keys = append(*keys, k)
 		}
 	}
-	sort.Strings(keys)
 
-	buf := keyGenPool.bufferPool.Get().(*bytes.Buffer)
+	// using a pointer to implementation of sort.Interface does not perform an alloc
+	sort.Sort((*sort.StringSlice)(keys))
```

Comment:
It seems that the Golang escape analysis has determined that the argument to `sort.Sort` escapes and thus, it seems that you cannot call `sort.Sort` with a stack-allocated value (as it has to escape). Therefore, I would recommend following @vdarulis suggestion of using an insertionsort on these. 

Using the insertionsort, I was also able to shave off ~16% of the CPU time:
```
# Command to test is 
$ go test -c -o keyPrefix.out ; ./keyPrefix.out -test.count=10 -test.run=asdfasdfasdf -test.bench="BenchmarkScopeTaggedCachedSubscopes" -test.benchmem -test.cpu 16 -test.memprofile=/home/user/mem.profile -test.cpuprofile=/home/user/cpu.profile  -test.mutexprofile=/home/user/mutex.profile -test.blockprofile=/home/user/block.profile > {before,after}.txt

$ benchstat before.txt after.txt                                                                                                      09:06 23/11
name                           old time/op    new time/op    delta
ScopeTaggedCachedSubscopes-16    1.06µs ± 2%    0.88µs ± 2%  -16.96%  (p=0.000 n=9+10)

name                           old alloc/op   new alloc/op   delta
ScopeTaggedCachedSubscopes-16      336B ± 0%      336B ± 0%     ~     (all equal)

name                           old allocs/op  new allocs/op  delta
ScopeTaggedCachedSubscopes-16      2.00 ± 0%      2.00 ± 0%     ~     (all equal)
```
(there were 10 runs and `before.txt` contains the numbers for this PR and `after.txt` contains the numbers for when the `[]string` value is stack-allocated and using an adapted form of https://www.golangprograms.com/golang-program-for-implementation-of-insertionsort.html) :

Reply:
yea thanks. I have removed sync.Pool and using stack allocated string slices and using insertionSort (copied from the std go sort pkg). sort.Sort() does do insertionSort internally when the len is <= 12